### PR TITLE
[IMP] sale_order_product_recommendation: slipt tests

### PR DESCRIPTION
**Pull request by chienandalu**
- Improve test declaration inherability

For cases like https://github.com/OCA/sale-workflow/pull/1005

cc @Tecnativa TT20511

### CODE REVIEW

File: sale_order_product_recommendation/tests/test_recommendation.py [hunk 1]
```diff
@@ -1,75 +1,13 @@
...
-
-from odoo.tests.common import SavepointCase
+from .test_recommendation_common import RecommendationCase
 from odoo.exceptions import UserError
 
 
-class RecommendationCase(SavepointCase):
+class RecommendationCaseTests(RecommendationCase):
     @classmethod
     def setUpClass(cls):
```
Comment (pedrobaeza) [inline sale_order_product_recommendation/tests/test_recommendation.py]: No need of keeping this method

File: sale_order_product_recommendation/tests/test_recommendation.py [hunk 2]
```diff
@@ -1,75 +1,13 @@
...
-                    "qty_delivered": 50,
-                }),
-            ],
-        })
-        # Create a new sale order for the same customer
-        cls.new_so = cls.env["sale.order"].create({
-            "partner_id": cls.partner.id,
-        })
 
     def wizard(self):
```
Comment (pedrobaeza) [inline sale_order_product_recommendation/tests/test_recommendation.py]: Isn't interesting to move this also to base?

File: sale_order_product_recommendation/tests/test_recommendation.py (hunk 1 above)
Comment (chienandalu) [reply sale_order_product_recommendation/tests/test_recommendation.py]: ok, thanks :)

File: sale_order_product_recommendation/tests/test_recommendation.py (hunk 2 above)
Comment (chienandalu) [reply sale_order_product_recommendation/tests/test_recommendation.py]: Yes, I noticed it the moment I started writing the extension :)

**Review by pedrobaeza — APPROVED** [review]
Fast-tracking this as it doesn't involve any change of logic.

**Pull request merged**